src/pages/resultsView/plots/PlotsTabSelectionHistory.ts

```diff
@@ -0,0 +1,93 @@
+import { AxisMenuSelection, PlotsTabGeneOption, PlotsTabOption } from "./PlotsTab";
+import autobind from "autobind-decorator";
+
+type Selection = {
+    gene?: PlotsTabGeneOption | undefined,
+    geneSet?: PlotsTabOption | undefined,
+    source?: PlotsTabOption | undefined,
+    treatment?: PlotsTabOption | undefined,
+
+}
+
+type FieldUpdater = (option: any) => void;
+
+export default class PlotsTabSelectionHistory {
+    private horizontal: Map<string, Selection> = new Map();
+    private vertical: Map<string, Selection> = new Map();
+
+    @autobind
+    public updateHorizontalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.horizontal, newSelection);
+    }
+    
+    @autobind
+    public updateVerticalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.vertical, newSelection);
+    }
+
+    private updateAxisWithSelection(axis: Map<string, Selection>, newSelection: AxisMenuSelection): void {
+        if (newSelection.dataType !== undefined) {
+            let selectionToUpdate = axis.get(newSelection.dataType)
+
+            if (selectionToUpdate === undefined) {
+                selectionToUpdate = {};
+            }
+
+            axis.set(newSelection.dataType, this.updateSelection(selectionToUpdate, newSelection))
+        }
+    }
+
+    private updateSelection(selectionToUpdate: Selection, newSelection: AxisMenuSelection): Selection {
+        selectionToUpdate.gene = newSelection.selectedGeneOption;
+        selectionToUpdate.geneSet = newSelection.selectedGenesetOption;
+        selectionToUpdate.source = newSelection.selectedDataSourceOption;
+        selectionToUpdate.treatment = newSelection.selectedTreatmentOption;
+        return selectionToUpdate;
+    }
+
+    @autobind
+    public updateHorizontalExternalSelection(
```

Comment:
what is this method for? Im confused by the external selection and the updaters thing. maybe more descriptive naming would help, maybe comments?

Reply:
I simplified the name a little bit and added some documentation. Hopefully that helps.